src/components/button/ProgressBar.js

```diff
@@ -0,0 +1,134 @@
+import React, { Component, createRef } from 'react';
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import { random } from 'lodash';
+import onTransitionEnd from 'proper-on-transition-end';
+import styles from './ProgressBar.css';
+
+const INCREMENT_INTERVAL = 300;
+const MAX_PROGRESS = 0.95;
+
+class ProgressBar extends Component {
+    progressBarRef = createRef();
+
+    componentDidMount() {
+        this.maybeHandleRunningChange(false);
```

Comment:
If I understood correctly, the following would be better to understand: `this.props.running && this.begin();`

Reply:
I'd prefer leaving as is, as the behavior is used multiple times and is already abstracted in a function. You'd have to consider the execution of `this.end()` as well, in case `running`starts off as false, and to have that if/else format, we might as well keep the function we already have.

Reply:
Seems fine to me 😁